### PR TITLE
metrics: Add remote cpu consumption test.

### LINKED ISSUE #763: Add remote cpu consumption test
Add remote cpu consumption test while running an iperf3 bandwidth test.

### PR DESCRIPTION
**Pull request by GabyCT**
Add remote cpu consumption test while running an iperf3 bandwidth test.

Fixes #763

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

**Comment by GabyCT** [comment]
@iphutch could you please review it thanks

File: metrics/network/remote_network/remote-networking-cpu.sh [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+		i)
+			interface_name="${OPTARG}"
+			;;
+		u)
+			ssh_user="${OPTARG}"
+			;;
+		\?)
+			echo "An invalid option has been entered: -$OPTARG";
+			help
+			exit 0;
```
Comment (grahamwhaley) [inline metrics/network/remote_network/remote-networking-cpu.sh]: Can we make this and the failure case below an `exit 1`?

File: metrics/network/remote_network/remote-networking-cpu.sh [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+		:)
+			echo "Missing argument for -$OPTARG";
+			help
+			exit 0;
+			;;
+		esac
+	done
+	shift $((OPTIND-1))
+
+	[[ -z "$interface_name" ]] && help && die "Missing IP Address."
```
Comment (grahamwhaley) [inline metrics/network/remote_network/remote-networking-cpu.sh]: Looks like you have the `interface` and `address` warnings swapped around

File: metrics/network/remote_network/remote-networking-cpu.sh [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+	[[ -z "$ssh_address" ]] && help && die "Missing Swarm Interface."
+	[[ -z "$ssh_user" ]] && help && die "Missing User."
+
+	if [ "$test_cpu" == "1" ]; then
```
Comment (grahamwhaley) [inline metrics/network/remote_network/remote-networking-cpu.sh]: As presently the only action this test has is to run the cpu_consumption test, I think you can remove the `-c` option and this test here, and as long as you have passed the checks above, always run the cpu test - you think @GabyCT ?

File: metrics/network/remote_network/remote-networking-cpu.sh [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+			;;
+		\?)
+			echo "An invalid option has been entered: -$OPTARG";
+			help
+			exit 0;
+			;;
+		:)
+			echo "Missing argument for -$OPTARG";
+			help
+			exit 0;
```
Comment (devimc) [inline metrics/network/remote_network/remote-networking-cpu.sh]: same here?

Comment (grahamwhaley) [reply metrics/network/remote_network/remote-networking-cpu.sh]: yep - hence my 'and the failure case below' ;-)

**Comment by GabyCT** [comment]
@grahamwhaley and @devimc let me work on the changes :)

**Comment by GabyCT** [comment]
@devimc and @grahamwhaley changes applied

File: metrics/network/remote_network/remote-networking-cpu.sh [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+		-p $(pidof ${process}) | awk '{ total+= $1 } END { print total/NR }')
+	echo "CPU % consumption: $total_cpu_consumption $units"
+	save_results "$test_name" "CPU consumption" "$total_cpu_consumption" "$units"
+
+	clean_environment
+}
+
+function main {
+	local OPTIND
+	while getopts ":a:h:i:u:" opt
```
Comment (MarioCarrilloA) [inline metrics/network/remote_network/remote-networking-cpu.sh]: does `h` option need a value?

**Comment by devimc** [comment]
lgtm

File: metrics/network/remote_network/remote-networking-cpu.sh [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+EOF
+)"
+}
+
+# This function will measure CPU consumption
+# will running an iperf3 bandwidth test
+function remote_network_cpu_consumption {
+	test_name="Remote Network CPU Consumption"
+	units="%"
+	get_runtime
```
Comment (grahamwhaley) [inline metrics/network/remote_network/remote-networking-cpu.sh]: Just a thought. I suspect we should have an `init_env` call at the start here somewhere to clean up the host docker environment before we begin (but what to do about the remote server...???).
https://github.com/clearcontainers/tests/blob/master/metrics/lib/common.bash#L105
I see there are no such calls for any of the remote tests - that is probably something I and @MarioCarrilloA should have picked up on - and hopefully we will soon get the time to write a 'guide to writing metrics', and maybe a template metrics file with these things in.
Given we don't have `init_env` in any of the other remote tests, I'll open an Issue for it.

**Comment by GabyCT** [comment]
@rcaballeromx could you please take a look? thanks

**Comment by GabyCT** [comment]
@grahamwhaley I added the function thanks

**Comment by grahamwhaley** [comment]
Thanks @GabyCT ! :-)
We had what looked like a swarm bats fail on the 17.04 CI, so I've kicked a re-run on that.

**Comment by devimc** [comment]
lgtm

File: metrics/network/remote_network/README.md [hunk 7]
```diff
@@ -54,3 +57,17 @@ $ cd metrics/network/remote_network
 $ bash remote-networking-memory-smem.sh "[options]" -i "<interface_name>" -u "<user>" -a "<ip_address>"
 
 ```
+## Running the `remote-networking-cpu.sh` test
+
+`remote-networking-cpu.sh` script should run in host A. The script needs the following inputs to run:
```
Comment (rcaballeromx) [inline metrics/network/remote_network/README.md]: Use imperative:
Run the `remote-networking-cpu.sh` script in host A.

File: metrics/network/remote_network/README.md [hunk 8]
```diff
@@ -7,12 +7,15 @@ container that will act as a client. The network bandwidth will be measured acro
...
+- The `remote-networking-cpu.sh` script measures the percentage of CPU consumption used while running
+the maximum network bandwidth with `iperf3`.
+
 ## Prerequisite
 
 An automatic login from host A (user A) to host B (user B) is needed. You must setup authentication 
 keys to do an ssh login without password.
-The `remote-networking-iperf3.sh` and `remote-networking-memory-smem.sh` scripts are only compatible with
-`docker 1.12.1`.
+The `remote-networking-iperf3.sh`, `remote-networking-memory-smem.sh` and `remote-networking-cpu.sh`
```
Comment (rcaballeromx) [inline metrics/network/remote_network/README.md]: Oxford comma missing before the 'and'

File: metrics/network/remote_network/remote-networking-cpu.sh [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+# This function describes how to use this script
+function help {
+echo "$(cat << EOF
+Usage: $0 "[options]"
+	Description:
+		This script measures CPU consumption. Simultaneously, the
+		script runs a server container and host B runs a client
+		container.
+		To run this script, provide the following inputs:
+		- Interface name where swarm will run.
```
Comment (rcaballeromx) [inline metrics/network/remote_network/remote-networking-cpu.sh]: Capitalize Swarm if you refer to the application. If you mean the noun, then add 'the' before it.

File: metrics/network/remote_network/remote-networking-test-common.sh [hunk 10]
```diff
@@ -59,6 +59,17 @@ function get_runtime {
 	fi
 }
 
+# This function will check the environment in host B
```
Comment (rcaballeromx) [inline metrics/network/remote_network/remote-networking-test-common.sh]: Do not use future tense when commenting code. Stick to present tense:
This function checks the environment in host B.

**Comment by GabyCT** [comment]
@rcaballeromx changes applied thanks

File: metrics/network/remote_network/README.md [hunk 11]
```diff
@@ -54,3 +57,17 @@ $ cd metrics/network/remote_network
 $ bash remote-networking-memory-smem.sh "[options]" -i "<interface_name>" -u "<user>" -a "<ip_address>"
 
 ```
+## Running the `remote-networking-cpu.sh` test
+
+Run the `remote-networking-cpu.sh` script in host A. The script needs the following inputs to run:
+- Use `-i` to specify the name of the interface where the swarm will run.
```
Comment (rcaballeromx) [inline metrics/network/remote_network/README.md]: Capitalize: Swarm

**Pull request merged**